### PR TITLE
Feature/1529 add daterange to store

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #1529 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by aaemnnosttv** [comment]
@googlebot 👋

**Comment by eugene-manuilov** [comment]
@googlebot wake up :D

**Comment by eugene-manuilov** [comment]
@felixarntz could you please help me to solve `cla/google` check issue?

File: assets/js/googlesitekit/datastore/user/date-range.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			return {
+				...state,
+				dateRange: payload.slug,
+			};
+		default:
+			// do nothing
+			break;
+	}
+
+	return state;
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/googlesitekit/datastore/user/date-range.js]: Done.

File: assets/js/googlesitekit/datastore/user/date-range.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ */
+import { createTestRegistry } from '../../../../../tests/js/utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/user date-range', () => {
+	describe( 'actions', () => {
+		describe( 'setDateRange', () => {
+			it( 'should require the date range slug param', () => {
+				expect( () => {
+					const registry = createTestRegistry();
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/googlesitekit/datastore/user/date-range.test.js]: Done.

**Comment by aaemnnosttv** [comment]
@googlebot !

**Pull request merged**